src/core/drive/form_submission.js

```diff
@@ -116,64 +116,106 @@ export class FormSubmission {
 
   // Fetch request delegate
 
-  prepareRequest(request) {
-    if (!request.isSafe) {
+  prepareRequest(fetchRequest) {
+    if (!fetchRequest.isSafe) {
       const token = getCookieValue(getMetaContent("csrf-param")) || getMetaContent("csrf-token")
       if (token) {
-        request.headers["X-CSRF-Token"] = token
+        fetchRequest.headers.set("X-CSRF-Token", token)
       }
     }
 
-    if (this.requestAcceptsTurboStreamResponse(request)) {
-      request.acceptResponseType(StreamMessage.contentType)
+    if (this.requestAcceptsTurboStreamResponse(fetchRequest)) {
+      fetchRequest.acceptResponseType(StreamMessage.contentType)
     }
   }
 
-  requestStarted(_request) {
+  requestStarted(fetchRequest) {
+    const formSubmission = this
+
     this.state = FormSubmissionState.waiting
     this.submitter?.setAttribute("disabled", "")
     this.setSubmitsWith()
     dispatch("turbo:submit-start", {
       target: this.formElement,
-      detail: { formSubmission: this }
+      detail: {
+        request: fetchRequest.request,
+        submitter: this.submitter,
+
+        get formSubmission() {
+          console.warn("`event.detail.formSubmission` is deprecated. Use `event.target`, `event.detail.submitter`, and `event.detail.request` instead")
```

Comment:
@afcapel @kevinmcconnell I realize now that the removal of TypeScript in anticipation of an 8.0 release means that a deprecation cycle as part of a minor release is no longer possible.

I'm curious of your thoughts here. Ultimately, we can remove the properties entirely and skip the `console.warn` calls.